app/module/drivers/input/input_pinnacle.c

```diff
@@ -0,0 +1,386 @@
+            return ret;
+        }
+    }
+
+    if (config->sensitivity > PINNACLE_SENSITIVITY_1X) {
+        ret = pinnacle_era_write(dev, PINNACLE_ERA_REG_TRACKING_ADC_CONFIG,
+                                 pinnacle_adc_sensitivity_reg_value(config->sensitivity));
+        if (ret < 0) {
+            LOG_ERR("Failed to set ADC sensitivity %d", ret);
+            return ret;
+        }
+    }
+
+    uint8_t feed_cfg2 = PINNACLE_FEED_CFG2_EN_IM;
+    if (config->no_taps) {
+        feed_cfg2 |= PINNACLE_FEED_CFG2_DIS_TAP;
+    }
+    if (config->rotate_90) {
+        feed_cfg2 |= PINNACLE_FEED_CFG2_ROTATE_90;
+    }
+    ret = pinnacle_write(dev, PINNACLE_FEED_CFG2, feed_cfg2);
+    if (ret < 0) {
+        LOG_ERR("can't write %d", ret);
+        return ret;
+    }
+    uint8_t feed_cfg1 = PINNACLE_FEED_CFG1_EN_FEED;
+    if (feed_cfg1) {
+        ret = pinnacle_write(dev, PINNACLE_FEED_CFG1, feed_cfg1);
+    }
+    if (ret < 0) {
+        LOG_ERR("can't write %d", ret);
+        return ret;
+    }
+
+    data->dev = dev;
+
+    pinnacle_clear_status(dev);
+
+    gpio_pin_configure_dt(&config->dr, GPIO_INPUT);
+    gpio_init_callback(&data->gpio_cb, pinnacle_gpio_cb, BIT(config->dr.pin));
+    ret = gpio_add_callback(config->dr.port, &data->gpio_cb);
+    if (ret < 0) {
+        LOG_ERR("Failed to set DR callback: %d", ret);
+        return -EIO;
+    }
+
+    k_work_init(&data->work, pinnacle_work_cb);
+
+    pinnacle_write(dev, PINNACLE_FEED_CFG1, feed_cfg1);
+
+    set_int(dev, true);
+
+    return 0;
+}
+
+#define PINNACLE_INST(n)                                                                           \
+    static struct pinnacle_data pinnacle_data_##n;                                                 \
+    static const struct pinnacle_config pinnacle_config_##n = {                                    \
+        .bus = COND_CODE_1(                                                                        \
+            DT_INST_ON_BUS(0, i2c), (I2C_DT_SPEC_INST_GET(0)),                                     \
+            (SPI_DT_SPEC_INST_GET(0,                                                               \
+                                  SPI_OP_MODE_MASTER | SPI_WORD_SET(8) | SPI_LINES_SINGLE |        \
+                                      SPI_TRANSFER_MSB | SPI_MODE_CPHA,                            \
+                                  0))),                                                            \
+        .rotate_90 = DT_INST_PROP(0, rotate_90),                                                   \
+        .sleep_en = DT_INST_PROP(0, sleep),                                                        \
+        .no_taps = DT_INST_PROP(0, no_taps),                                                       \
+        .sensitivity = DT_INST_ENUM_IDX_OR(0, sensitivity, PINNACLE_SENSITIVITY_1X),               \
+        .dr = GPIO_DT_SPEC_GET_OR(DT_DRV_INST(0), dr_gpios, {}),                                   \
```

Comment:
I believe some of the `0`s in this macro should actually be `n`s, because we don't want to hardcode just one instance.